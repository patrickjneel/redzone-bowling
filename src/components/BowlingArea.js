import React from 'react';
import ScoreBoard from './ScoreBoard/ScoreBoard';

const BowlingArea = () => {
  return (
    <div>
      <ScoreBoard />
      Bowling things go here
    </div>
  )
}

export default BowlingArea;
