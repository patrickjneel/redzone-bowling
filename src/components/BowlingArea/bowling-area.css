.bowling-area {
  align-items: center;
  background-color: #FFFBF7;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;
}

.bowling-score-area {
  align-items: flex-end;
  display: flex;
  height: 400px;
  justify-content: center;
  width: 80%;
}

.input-and-btn-area {
  display: flex;
  justify-content: center;
  padding-bottom: 30px;
  width: 100%;
}
