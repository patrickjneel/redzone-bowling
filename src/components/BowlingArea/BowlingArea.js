import React, { Component } from 'react';
import ScoreBoard from '../ScoreBoard/ScoreBoard';
import './bowling-area.css';

class BowlingArea extends Component {
  constructor() {
    super()
    this.state = {
      playerName: '',
      scoreArr: []
    }
  }

  handleInput = (event) => {
   this.setState({
      playerName: event.target.value
    });
  };

  submitScore = (event) => {
    // if(typeof event.target.value !== 'number' ) {
    //   return (
    //     <div>Pleae enter a number</div>
    //   )
    // }
    this.state.scoreArr.push(event.target.value)
    // this.setState({ scoreArr: scores })
  }

  render() {
    console.log(this.state.scoreArr)
    return (
      <div className="bowling-area">
        <ScoreBoard name={this.state.playerName}/>
        <div className="bowling-score-area">
          <div className="input-and-btn-area">
            <input 
              className="player-inputs"
              placeholder="Player 1 Enter Name"
              value={this.state.playerName}
              onChange={this.handleInput}
            />
            <input 
              className="player-inputs"
              placeholder="Roll Away!!"
              value={this.state.scoreArr}
              onChange={this.submitScore}
            />
            <button onClick={this.submitScore}>
              Submit Score
            </button>
          </div>
        </div>
      </div>
    )
  }
}

export default BowlingArea;
