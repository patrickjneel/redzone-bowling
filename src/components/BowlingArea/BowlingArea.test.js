import React from 'react';
import { shallow } from 'enzyme';
import BowlingArea from './BowlingArea';

it('should match the snap shot', () => {
    const renderedBowlingArea = shallow(<BowlingArea />);
    expect(renderedBowlingArea).toMatchSnapshot();

});

it('should have a default state', () => {
  const renderedBowlingArea = shallow(<BowlingArea />);
  const expectedInitialPins = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
  const expectedRemainingPins = [];
  const expectedScoreArr = [];
  const expectedFrameTotal = [];
  const expectedScoreObj = {
    10: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    9: [0, 1],
    8: [0, 1, 2],
    7: [0, 1, 2, 3],
    6: [0, 1, 2, 3, 4],
    5: [0, 1, 2, 3, 4, 5],
    4: [0, 1, 2, 3, 4, 5, 6],
    3: [0, 1, 2, 3, 4, 5, 6, 7],
    2: [0, 1, 2, 3, 4, 5, 6, 7, 8],
    1: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
    0: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10] 
  };

  expect(renderedBowlingArea.state('initialPins')).toEqual(expectedInitialPins);
  expect(renderedBowlingArea.state('remainingPins')).toEqual(expectedRemainingPins);
  expect(renderedBowlingArea.state('scoreArr')).toEqual(expectedScoreArr);
  expect(renderedBowlingArea.state('frameTotal')).toEqual(expectedFrameTotal);
  expect(renderedBowlingArea.state('scoreObj')).toEqual(expectedScoreObj);

})
