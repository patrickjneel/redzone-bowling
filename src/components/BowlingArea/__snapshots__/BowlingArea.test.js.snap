// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snap shot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BowlingArea />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="button-container"
        >
          <button
            className="score-btn"
            onClick={[Function]}
            value={0}
          >
            0
          </button>
          <button
            className="score-btn"
            onClick={[Function]}
            value={1}
          >
            1
          </button>
          <button
            className="score-btn"
            onClick={[Function]}
            value={2}
          >
            2
          </button>
          <button
            className="score-btn"
            onClick={[Function]}
            value={3}
          >
            3
          </button>
          <button
            className="score-btn"
            onClick={[Function]}
            value={4}
          >
            4
          </button>
          <button
            className="score-btn"
            onClick={[Function]}
            value={5}
          >
            5
          </button>
          <button
            className="score-btn"
            onClick={[Function]}
            value={6}
          >
            6
          </button>
          <button
            className="score-btn"
            onClick={[Function]}
            value={7}
          >
            7
          </button>
          <button
            className="score-btn"
            onClick={[Function]}
            value={8}
          >
            8
          </button>
          <button
            className="score-btn"
            onClick={[Function]}
            value={9}
          >
            9
          </button>
          <button
            className="score-btn"
            onClick={[Function]}
            value={10}
          >
            10
          </button>
        </div>,
        undefined,
        <ScoreBoard
          frameTotal={Array []}
          scoreArray={Array []}
        />,
      ],
      "className": "bowling-area",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="score-btn"
              onClick={[Function]}
              value={0}
            >
              0
            </button>,
            <button
              className="score-btn"
              onClick={[Function]}
              value={1}
            >
              1
            </button>,
            <button
              className="score-btn"
              onClick={[Function]}
              value={2}
            >
              2
            </button>,
            <button
              className="score-btn"
              onClick={[Function]}
              value={3}
            >
              3
            </button>,
            <button
              className="score-btn"
              onClick={[Function]}
              value={4}
            >
              4
            </button>,
            <button
              className="score-btn"
              onClick={[Function]}
              value={5}
            >
              5
            </button>,
            <button
              className="score-btn"
              onClick={[Function]}
              value={6}
            >
              6
            </button>,
            <button
              className="score-btn"
              onClick={[Function]}
              value={7}
            >
              7
            </button>,
            <button
              className="score-btn"
              onClick={[Function]}
              value={8}
            >
              8
            </button>,
            <button
              className="score-btn"
              onClick={[Function]}
              value={9}
            >
              9
            </button>,
            <button
              className="score-btn"
              onClick={[Function]}
              value={10}
            >
              10
            </button>,
          ],
          "className": "button-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": 0,
              "className": "score-btn",
              "onClick": [Function],
              "value": 0,
            },
            "ref": null,
            "rendered": 0,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": 1,
              "className": "score-btn",
              "onClick": [Function],
              "value": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": 2,
              "className": "score-btn",
              "onClick": [Function],
              "value": 2,
            },
            "ref": null,
            "rendered": 2,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": 3,
              "className": "score-btn",
              "onClick": [Function],
              "value": 3,
            },
            "ref": null,
            "rendered": 3,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "children": 4,
              "className": "score-btn",
              "onClick": [Function],
              "value": 4,
            },
            "ref": null,
            "rendered": 4,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "host",
            "props": Object {
              "children": 5,
              "className": "score-btn",
              "onClick": [Function],
              "value": 5,
            },
            "ref": null,
            "rendered": 5,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "host",
            "props": Object {
              "children": 6,
              "className": "score-btn",
              "onClick": [Function],
              "value": 6,
            },
            "ref": null,
            "rendered": 6,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "7",
            "nodeType": "host",
            "props": Object {
              "children": 7,
              "className": "score-btn",
              "onClick": [Function],
              "value": 7,
            },
            "ref": null,
            "rendered": 7,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "8",
            "nodeType": "host",
            "props": Object {
              "children": 8,
              "className": "score-btn",
              "onClick": [Function],
              "value": 8,
            },
            "ref": null,
            "rendered": 8,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "9",
            "nodeType": "host",
            "props": Object {
              "children": 9,
              "className": "score-btn",
              "onClick": [Function],
              "value": 9,
            },
            "ref": null,
            "rendered": 9,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "10",
            "nodeType": "host",
            "props": Object {
              "children": 10,
              "className": "score-btn",
              "onClick": [Function],
              "value": 10,
            },
            "ref": null,
            "rendered": 10,
            "type": "button",
          },
        ],
        "type": "div",
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "frameTotal": Array [],
          "scoreArray": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="button-container"
          >
            <button
              className="score-btn"
              onClick={[Function]}
              value={0}
            >
              0
            </button>
            <button
              className="score-btn"
              onClick={[Function]}
              value={1}
            >
              1
            </button>
            <button
              className="score-btn"
              onClick={[Function]}
              value={2}
            >
              2
            </button>
            <button
              className="score-btn"
              onClick={[Function]}
              value={3}
            >
              3
            </button>
            <button
              className="score-btn"
              onClick={[Function]}
              value={4}
            >
              4
            </button>
            <button
              className="score-btn"
              onClick={[Function]}
              value={5}
            >
              5
            </button>
            <button
              className="score-btn"
              onClick={[Function]}
              value={6}
            >
              6
            </button>
            <button
              className="score-btn"
              onClick={[Function]}
              value={7}
            >
              7
            </button>
            <button
              className="score-btn"
              onClick={[Function]}
              value={8}
            >
              8
            </button>
            <button
              className="score-btn"
              onClick={[Function]}
              value={9}
            >
              9
            </button>
            <button
              className="score-btn"
              onClick={[Function]}
              value={10}
            >
              10
            </button>
          </div>,
          undefined,
          <ScoreBoard
            frameTotal={Array []}
            scoreArray={Array []}
          />,
        ],
        "className": "bowling-area",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="score-btn"
                onClick={[Function]}
                value={0}
              >
                0
              </button>,
              <button
                className="score-btn"
                onClick={[Function]}
                value={1}
              >
                1
              </button>,
              <button
                className="score-btn"
                onClick={[Function]}
                value={2}
              >
                2
              </button>,
              <button
                className="score-btn"
                onClick={[Function]}
                value={3}
              >
                3
              </button>,
              <button
                className="score-btn"
                onClick={[Function]}
                value={4}
              >
                4
              </button>,
              <button
                className="score-btn"
                onClick={[Function]}
                value={5}
              >
                5
              </button>,
              <button
                className="score-btn"
                onClick={[Function]}
                value={6}
              >
                6
              </button>,
              <button
                className="score-btn"
                onClick={[Function]}
                value={7}
              >
                7
              </button>,
              <button
                className="score-btn"
                onClick={[Function]}
                value={8}
              >
                8
              </button>,
              <button
                className="score-btn"
                onClick={[Function]}
                value={9}
              >
                9
              </button>,
              <button
                className="score-btn"
                onClick={[Function]}
                value={10}
              >
                10
              </button>,
            ],
            "className": "button-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": 0,
                "className": "score-btn",
                "onClick": [Function],
                "value": 0,
              },
              "ref": null,
              "rendered": 0,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": 1,
                "className": "score-btn",
                "onClick": [Function],
                "value": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": 2,
                "className": "score-btn",
                "onClick": [Function],
                "value": 2,
              },
              "ref": null,
              "rendered": 2,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "children": 3,
                "className": "score-btn",
                "onClick": [Function],
                "value": 3,
              },
              "ref": null,
              "rendered": 3,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "host",
              "props": Object {
                "children": 4,
                "className": "score-btn",
                "onClick": [Function],
                "value": 4,
              },
              "ref": null,
              "rendered": 4,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "5",
              "nodeType": "host",
              "props": Object {
                "children": 5,
                "className": "score-btn",
                "onClick": [Function],
                "value": 5,
              },
              "ref": null,
              "rendered": 5,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "6",
              "nodeType": "host",
              "props": Object {
                "children": 6,
                "className": "score-btn",
                "onClick": [Function],
                "value": 6,
              },
              "ref": null,
              "rendered": 6,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "7",
              "nodeType": "host",
              "props": Object {
                "children": 7,
                "className": "score-btn",
                "onClick": [Function],
                "value": 7,
              },
              "ref": null,
              "rendered": 7,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "8",
              "nodeType": "host",
              "props": Object {
                "children": 8,
                "className": "score-btn",
                "onClick": [Function],
                "value": 8,
              },
              "ref": null,
              "rendered": 8,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "9",
              "nodeType": "host",
              "props": Object {
                "children": 9,
                "className": "score-btn",
                "onClick": [Function],
                "value": 9,
              },
              "ref": null,
              "rendered": 9,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "10",
              "nodeType": "host",
              "props": Object {
                "children": 10,
                "className": "score-btn",
                "onClick": [Function],
                "value": 10,
              },
              "ref": null,
              "rendered": 10,
              "type": "button",
            },
          ],
          "type": "div",
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "frameTotal": Array [],
            "scoreArray": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
